--- a/gettext-runtime/po/Makefile.in.in	2026-05-02 09:45:41.695298043 -0500
+++ b/gettext-runtime/po/Makefile.in.in	2026-05-02 09:46:11.498102071 -0500
@@ -476,3 +476,9 @@
 # Tell versions [3.59,3.63) of GNU make not to export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
+
+# This Makefile contains rules which don't work with parallel make,
+# namely dist2.
+# See <https://lists.gnu.org/archive/html/bug-gettext/2022-06/msg00022.html>.
+# So, turn off parallel execution in this Makefile.
+.NOTPARALLEL:
