--- grub2-1.96+20080724.orig/util/grub.d/10_linux.in	2008-06-26 09:03:26.000000000 +0200
+++ grub2-1.96+20080724/util/grub.d/10_linux.in	2008-08-07 16:30:31.000000000 +0200
@@ -69,16 +69,7 @@
 {
   local a=`echo $1 | sed -e "s/vmlinu[zx]-//g"`
   local b=`echo $2 | sed -e "s/vmlinu[zx]-//g"`
-  local cmp=gt
-  if [ "x$b" = "x" ] ; then
-    return 0
-  fi
-  case $a:$b in
-    *.old:*.old) ;;
-    *.old:*) a=`echo -n $a | sed -e s/\.old$//g` ; cmp=gt ;;
-    *:*.old) b=`echo -n $b | sed -e s/\.old$//g` ; cmp=ge ;;
-  esac
-  test_numeric $a $cmp $b
+  dpkg --compare-versions "$1" gt "$2"
   return $?
 }
 
