
Workaround untill #484297 is fixed.

diff -ur grub2.old/util/grub.d/10_linux.in grub2/util/grub.d/10_linux.in
--- grub2.old/util/grub.d/10_linux.in	2008-06-17 01:00:28.000000000 +0200
+++ grub2/util/grub.d/10_linux.in	2008-06-17 01:06:53.000000000 +0200
@@ -27,7 +27,8 @@
   OS="${GRUB_DISTRIBUTOR} GNU/Linux"
 fi
 
-if [ "x${GRUB_DEVICE_UUID}" = "x" ] || [ "x${GRUB_DISABLE_LINUX_UUID}" = "xtrue" ] ; then
+if [ "x${GRUB_DEVICE_UUID}" = "x" ] || [ "x${GRUB_DISABLE_LINUX_UUID}" = "xtrue" ] \
+  || [ "`grub-probe -t abstraction --device ${GRUB_DEVICE}`" != "" ] ; then
   # loop-AES arranges things so that /dev/loop/X can be our root device, but
   # the initrds that Linux uses don't like that.
   case ${GRUB_DEVICE} in
