Index: b/util/setup.c
===================================================================
--- a/util/setup.c
+++ b/util/setup.c
@@ -58,6 +58,8 @@
 
 #include <errno.h>
 
+#define CORE_IMG_IN_FS		"setup_left_core_image_in_filesystem"
+
 /* On SPARC this program fills in various fields inside of the 'boot' and 'core'
  * image files.
  *
@@ -590,6 +592,8 @@
 
     grub_free (sectors);
 
+    unlink (DEFAULT_DIRECTORY "/" CORE_IMG_IN_FS);
+
     goto finish;
   }
 
@@ -632,6 +636,10 @@
   /* The core image must be put on a filesystem unfortunately.  */
   grub_util_info ("will leave the core image on the filesystem");
 
+  fp = grub_util_fd_open (DEFAULT_DIRECTORY "/" CORE_IMG_IN_FS,
+			  GRUB_UTIL_FD_O_WRONLY);
+  grub_util_fd_close (fp);
+
   grub_util_biosdisk_flush (root_dev->disk);
 
   /* Clean out the blocklists.  */
