
We like our blue theme.  Okuji doesn't.  ;-)

diff -x CVS -ur grub2/normal/menu.c grub2.color/normal/menu.c
--- grub2/normal/menu.c	2007-12-25 12:11:18.000000000 +0100
+++ grub2.color/normal/menu.c	2007-12-25 12:12:08.000000000 +0100
@@ -25,8 +25,8 @@
 #include <grub/env.h>
 #include <grub/script.h>
 
-#define GRUB_COLOR_MENU_NORMAL		0x07
-#define GRUB_COLOR_MENU_HIGHLIGHT	0x70
+#define GRUB_COLOR_MENU_NORMAL		0x13
+#define GRUB_COLOR_MENU_HIGHLIGHT	0x1f
 
 static void
 draw_border (void)
